bin/cli.js

```diff
@@ -9,75 +10,96 @@ const yargs = require('yargs');
 const PWMetrics = require('../lib/index');
 const { getConfigFromFile } = require('../lib/utils/fs');
 const { getMessageWithPrefix } = require('../lib/utils/messages');
+const { mergeDeep } = require('../lib/utils/object');
 
 const cliFlags = yargs
   .help('help')
   .version(() => require('../package').version)
   .showHelpOnFail(false, 'Specify --help for available options')
-
-  .usage('$0 url')
-
-  .describe({
-    'json': 'Reports json details to stdout',
-    'runs': 'Does n runs (eg. 3, 5), and reports the median run\'s numbers.',
-    'expectations': 'Expectations from metrics results. Useful for CI. Config required.',
-    'submit': 'Submit metric results into sheets. Config required.',
-    'config': 'Read config form file.',
-    'disable-cpu-throttling': 'Disable CPU throttling',
-    'and more Lighthouse CLI options': 'Docs - https://github.com/GoogleChrome/lighthouse/#lighthouse-cli-options',
+  .wrap(yargs.terminalWidth())
+  .usage('Usage: $0 <url>')
+  .command('url', 'URL to test')
+  .option('json', {
+    'describe': 'JSON output defaults to stdout.',
+    'type': 'boolean',
+    'default': false,
+    'group': 'Output:'
   })
-  .default('disable-cpu-throttling', false)
-  .default('runs', 1)
-
-  // boolean values
-  .boolean([
-    'disable-cpu-throttling'
-  ])
+  .option('output-path', {
+    'describe': 'The file path to output the results',
+    'type': 'string',
+    'default': '',
+    'group': 'Output:'
+  })
+  .option('runs', {
+    'describe': 'Does n runs (eg. 3, 5), and reports the median run\'s numbers.',
+    'type': 'number',
+    'default': 1,
+  })
+  .option('expectations', {
+    'describe': 'Expectations from metrics results. Useful for CI. Config required.',
+    'type': 'boolean',
+    'default': false
+  })
+  .option('submit', {
+    'describe': 'Submit metric results into google sheets. Config required.',
+    'type': 'boolean',
+    'default': false
+  })
+  .option('config', {
+    'describe': 'Path to json config file',
+    'coerce': (arg) => getConfigFromFile(arg),
+    'type': 'string'
+  })
+  .option('disable-cpu-throttling', {
+    'describe': 'Disable CPU throttling',
+    'type': 'boolean',
+    'default': false
+  })
+  .epilogue('For more Lighthouse CLI options see https://github.com/GoogleChrome/lighthouse/#lighthouse-cli-options')
   .check(argv => {
-    if (argv._.length === 0)
-      throw new Error(getMessageWithPrefix('ERROR', 'NO_URL'));
+    //test possible url locations, first from cmd line then from config file
+    if (argv._.length !== 0 || (argv.config !== undefined && argv.config.url))
```

Comment:
It's kinda pattern/way for handling errors. Here some article -http://www.codingdefined.com/2015/10/what-are-error-first-callbacks-in-nodejs.html